objc2/src/macros.rs

```diff
@@ -58,6 +59,61 @@ macro_rules! sel {
     });
 }
 
+/// TODO
+#[cfg(feature = "static-sel")]
+#[macro_export]
+macro_rules! sel {
+    ($name:ident) => {
+        $crate::sel!(@__inner concat!(stringify!($name), '\0'), $name)
+    };
+    ($($name:ident :)+) => {
+        $crate::sel!(@__inner concat!($(stringify!($name), ':'),+, '\0'), $($name),+)
+    };
+    (@__inner $sel:expr, $($idents:ident),+) => {{
+        // HACK: Wrap the statics in a non-generic, `#[inline(never)]`
+        // function to "coerce" the compiler to group all of them into the
+        // same codegen unit to avoid link errors
+        #[inline(never)]
+        fn objc_static_workaround() -> $crate::runtime::Sel {
+            const X: &[u8] = $sel.as_bytes();
+
+            #[link_section = "__DATA,__objc_imageinfo,regular,no_dead_strip"]
+            #[export_name = concat!("\x01L_OBJC_IMAGE_INFO_", $crate::__proc_macros::hash_idents!($($idents)+))]
+            #[used]
+            static _IMAGE_TAG: [u32; 2] = [0, 0];
+
+            #[link_section = "__TEXT,__objc_methname,cstring_literals"]
+            #[export_name = concat!("\x01L_OBJC_METH_VAR_NAME_", $crate::__proc_macros::hash_idents!($($idents)+))]
+            static NAME: [u8; X.len()] = {
+                let mut res: [u8; X.len()] = [0; X.len()];
+                let mut i = 0;
+                while i < X.len() {
+                    res[i] = X[i];
+                    i += 1;
+                }
+                res
+            };
+
+            // Place the constant value in the correct section.
+            #[link_section = "__DATA,__objc_selrefs,literal_pointers"]
+            #[export_name = concat!("\x01L_OBJC_SELECTOR_REFERENCES_", $crate::__proc_macros::hash_idents!($($idents)+))]
+            static mut REF: &[u8; X.len()] = &NAME;
+
+            // The actual selector is replaced by dyld when the program is
+            // loaded, so we need to use a volatile read to prevent the
+            // optimizer from thinking it can circumvent the read through REF.
+            //
+            // TODO: `::core` here could be replaced with some more
+            // sophisticated logic so we don't rely on downstream users having
+            // this setup.
```

Comment:
You can re-export `core` in your crate and then access it from this macro as e.g. `$crate::_core`.

Reply:
Yeah I know, I was just carrying over a TODO from previous authors. Thanks though!